code/game/mecha/equipment/tools/tools.dm

```diff
@@ -49,7 +50,7 @@
 			if(locate(/mob/living) in O)
 				occupant_message(SPAN_WARNING("You can't load living things into the cargo compartment."))
 				return
-			if(O.anchored)
+			if(O.anchored &! allow_pickup)
```

Comment:
You are attempting to bitwise AND a list() with a boolean value.

Here's some quick half-assed code to showcase *A* method to accomplish what you want.
```
if(O.anchored && (!istype(O, /obj/structure/scrap) && !istype(O, /obj/structure/salvageable))
```

Alternatively, just get out of your mech and unanchor the scrap, if that's even possible?

Reply:
Thx, tryed to do that at the start and kinda failed at it